Node/Package managers guidelines.md

```diff
@@ -0,0 +1,200 @@
+
+**Understanding Doppelgangers in npm and Yarn**
+
+When using npm or yarn, dependencies are resolved and installed in a way that can lead to multiple versions of the same package being installed in different parts of the node\_modules structure. This happens because each package in the tree may have its own specific version of a dependency that doesn’t match other versions requested by different packages.
+
+For example:
+
+* Package A requires `lodash@4.17.0`.
+* Package B requires `lodash@4.16.0`.
+
+In npm or yarn, both versions may be installed separately, leading to redundant copies of lodash scattered across the project. This duplication can waste disk space, slow down installations, and lead to inconsistencies, especially if different versions of the same package behave differently.
+
+**Why We Chose pnpm to avoid Doppelgangers**
+
+pnpm takes a fundamentally different approach to dependency management that directly addresses the doppelganger problem. Instead of installing dependencies in a flat or deeply nested structure, pnpm uses a content-addressable storage system where all versions of a package are stored globally on the filesystem and linked into your project’s node\_modules using hard links or symlinks.
+
+This approach offers several key advantages:
+
+* **No Redundant Installations**: With pnpm, different versions of the same package are never redundantly installed. If a version of a package is already present in the global store, it is simply linked to where it’s needed. This eliminates the doppelganger issue entirely.
+* **Consistent Dependency Resolution**: pnpm enforces a strict dependency tree that ensures only the correct and intended versions of packages are used. This minimizes the risk of conflicts that can arise from duplicate installations.
+* **Efficient Disk Usage**: By storing each version of a package only once and linking it as needed, pnpm drastically reduces the disk space used by node modules, even in large projects with many dependencies.
+* **Improved Performance**: Because pnpm avoids redundant downloads and installations, it’s faster to install dependencies, especially in projects with complex dependency trees.
+
+### Getting Started with pnpm
+
+To start using pnpm, you first need to install it globally:
+
+```bash
+npm install -g pnpm
+```
+
+You can then use pnpm just like any other package manager, but with some enhancements.
+
+#### Installing Dependencies
+
+To install all dependencies listed in your package.json, run:
+
+```bash
+pnpm install
+```
+
+#### Adding a New Package
```

Comment:
Maybe describe some frequently used flags, like `prod` and `frozen-lockfile`.

Reply:
Added more commands